image-rs/src/unpack.rs

```diff
@@ -33,6 +33,11 @@ pub fn unpack<R: io::Read>(input: R, destination: &Path) -> Result<()> {
     let mut dirs: HashMap<CString, [timeval; 2]> = HashMap::default();
     for file in archive.entries()? {
         let mut file = file?;
+
+        if ! convert_whiteout(&file.path().unwrap(), file.header(), destination)? {
```

Comment:
Let's try to avoid using `unwrap()` here as it will cause the program panic. Try this `ok_or(anyhow!(...))` to raise a proper error information.

Reply:
Thanks -- total work in progress here :)